web-gateway/app/controllers/Currency.java

```diff
@@ -77,4 +77,9 @@ public boolean isValidStep(double value) {
         int price = toPriceUnits(value);
         return price % step == 0;
     }
+    
+    public boolean isValidIncrement(double value) {
+    	int price = (int) (value * 2);
+    	return price > 0 && price < 100 * Math.pow(10, 2 - currency.getDefaultFractionDigits());
+    }
```

Comment:
How about:

```java
int price = toPriceUnits(value);
return price > 0 && price < 100 * step;
```

In US dollars, this makes it 0 < price < $50, and in Yen this makes it 0 < price < 5000.